core/src/main/java/org/infinispan/stats/impl/AbstractClusterStats.java

```diff
@@ -0,0 +1,162 @@
+package org.infinispan.stats.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.jmx.JmxStatisticsExposer;
+import org.infinispan.jmx.annotations.DataType;
+import org.infinispan.jmx.annotations.DisplayType;
+import org.infinispan.jmx.annotations.ManagedAttribute;
+import org.infinispan.jmx.annotations.ManagedOperation;
+import org.infinispan.jmx.annotations.Parameter;
+import org.infinispan.jmx.annotations.Units;
+import org.infinispan.util.TimeService;
+import org.infinispan.util.logging.Log;
+
+/**
+ * @author Ryan Emerson
+ * @since 9.0
+ */
+public abstract class AbstractClusterStats implements JmxStatisticsExposer {
+
+   public static final long DEFAULT_STALE_STATS_THRESHOLD = 3000;
+
+   protected TimeService timeService;
```

Comment:
it seems the ```timeService``` is used to reset ```resetNanoseconds```.  I guess that can be done in a start method instead of in the injecting dependencies method.